src/app/shared/issue-tables/issue-sorter.ts

```diff
@@ -0,0 +1,39 @@
+import { Issue, ISSUE_TYPE_ORDER, SEVERITY_ORDER } from '../../core/models/issue.model';
+import { MatSort, SortDirection } from '@angular/material';
+
+export function getSortedData(sort: MatSort, data: Issue[]): Issue[] {
+  if (!sort.active) {
+      return data;
+  }
+  return data.sort((a, b) => {
+    switch (sort.active) {
+      case 'type':
+        return compareValue(sort.direction, ISSUE_TYPE_ORDER[a.type], ISSUE_TYPE_ORDER[b.type]);
```

Comment:
Shall we take this opportunity to refactor this method further?
Suggestion: Create separate methods for each sort key (i.e. `compareByType`, `compareBySeverity`)
I think the method will become more readable (reader can then view the sorting logic at a high level, without looking at details)

I also believe `compareValue` should not take `sort.direction` as a parameter.
We can obtain the result from `compareValue`(in this switch statement) and then flip it once based on `sort.direction`.
No need to handle `sort.direction` within the body of `compareValue` / `compareIntegerValue`

Reply:
Alright, so now we do not need to pass the `sort.direction` to the compare methods! 

I assume we can have other methods by `SEVERITY_ORDER` and `ISSUE_TYPE_ORDER`! Would there need to be other methods? I assume the rest can be compared easily by lexographical order or by integer value due to the nature of the other attributes!

Reply:
Sure, let's leave the other sort keys unchanged (besides removing `sort.direction` as an argument)

Reply:
👍